arduino-ide-extension/src/browser/contributions/board-selection.ts

```diff
@@ -79,22 +82,80 @@ export class BoardSelection extends SketchContribution {
           );
           return;
         }
-        const boardDetails = await this.boardsService.getBoardDetails({
-          fqbn: selectedBoard.fqbn,
-        });
-        if (boardDetails) {
-          const { VID, PID } = boardDetails;
-          const detail = `BN: ${selectedBoard.name}
-VID: ${VID}
-PID: ${PID}`;
-          await remote.dialog.showMessageBox(remote.getCurrentWindow(), {
-            message: nls.localize('arduino/board/boardInfo', 'Board Info'),
-            title: nls.localize('arduino/board/boardInfo', 'Board Info'),
-            type: 'info',
-            detail,
-            buttons: [nls.localize('vscode/issueMainService/ok', 'OK')],
-          });
+        // IDE2 must show the board info based on the selected port.
+        // https://github.com/arduino/arduino-ide/issues/1489
+
+        // IDE 1.x does not support network ports
+        if (selectedPort.protocol === 'network') {
+          this.messageService.info(
+            nls.localize(
+              'arduino/board/noNetworkPort',
+              "Network port, can't obtain info."
+            )
+          );
+          return;
+        }
+        const selectedPortKey = Port.keyOf(selectedPort);
+        console.log('selectedportkey', JSON.stringify(selectedPortKey));
+        console.log(
+          'availableboards',
+          JSON.stringify(
+            this.boardsServiceProvider.availableBoards.filter(
+              AvailableBoard.hasPort
+            )
+          )
+        );
+
+        const state = await this.boardsService.getState();
+        const boardListOnSelectedPort = Object.entries(state).filter(
+          ([portKey, [, boards]]) =>
+            portKey === selectedPortKey && boards.length
+        );
+        // IDE 1.x show this when cannot identify for example a ESP8266MOD although compile and upload works
+        if (!boardListOnSelectedPort.length) {
+          this.messageService.info(
+            nls.localize(
+              'arduino/board/noNativeSerialPort',
+              "Native serial port, can't obtain info."
+            )
+          );
+          return;
```

Comment:
There are three possible cases:

- identified port
- unidentified port
- native port

An example of all three is shown here:

```text
$ arduino-cli board list --format json
[
  {
    "port": {
      "address": "/dev/ttyUSB0",
      "label": "/dev/ttyUSB0",
      "protocol": "serial",
      "protocol_label": "Serial Port (USB)",
      "properties": {
        "pid": "0x6001",
        "serialNumber": "AB0P8H9K",
        "vid": "0x0403"
      }
    }
  },
  {
    "matching_boards": [
      {
        "name": "Arduino Uno",
        "fqbn": "arduino:avr:uno"
      }
    ],
    "port": {
      "address": "/dev/ttyACM0",
      "label": "/dev/ttyACM0",
      "protocol": "serial",
      "protocol_label": "Serial Port (USB)",
      "properties": {
        "pid": "0x0043",
        "serialNumber": "95032303537351D0F001",
        "vid": "0x2341"
      }
    }
  },
  {
    "port": {
      "address": "/dev/ttyS4",
      "label": "/dev/ttyS4",
      "protocol": "serial",
      "protocol_label": "Serial Port"
    }
  }
]
```

## Identified

`/dev/ttyACM0` is an identified port:

**Arduino IDE 1.x**

![image](https://user-images.githubusercontent.com/8572152/211132280-24a3f727-3ac6-4f02-86bd-eea215a907ec.png)

**Arduino IDE 2.x**

![image](https://user-images.githubusercontent.com/8572152/211132368-cccda3f1-434e-416b-a41e-d71e646cb0eb.png)

🙂 Arduino IDE 2.x behavior matches 1.x

## Unidentified

`/dev/ttyUSB0` is an unidentified port (it is a [classic **Nano**](https://store.arduino.cc/products/arduino-nano), which uses the manufacturer provided VID/PID pair of its general purpose FTDI 232RL USB chip, and thus can't be identified as a specific board):

**Arduino IDE 1.x**

![image](https://user-images.githubusercontent.com/8572152/211132447-12c663e2-241e-4749-8a93-649c84b221a6.png)

**Arduino IDE 2.x**

![image](https://user-images.githubusercontent.com/8572152/211132463-6b875def-3dcb-4376-b3b1-c0cd59a4edba.png)

🐛 Arduino IDE 2.x behavior does not match 1.x

The behavior of the current PR code is problematic for two reasons:

- It is incorrect to call `/dev/ttyUSB0` a "native port".
- It makes it so the VID and PID (which is valuable information for support/troubleshooting) can not be determined.

## Native

`/dev/ttyS4` is a native port:

**Arduino IDE 1.x**

![image](https://user-images.githubusercontent.com/8572152/211132488-193d965a-8595-45c7-8904-e280df934218.png)

**Arduino IDE 2.x**

![image](https://user-images.githubusercontent.com/8572152/211132463-6b875def-3dcb-4376-b3b1-c0cd59a4edba.png)

🙂 Arduino IDE 2.x behavior matches 1.x